explainability-core/src/main/java/org/kie/trustyai/explainability/model/Dataframe.java

```diff
@@ -46,23 +47,28 @@ public class Dataframe {
     private final List<List<Value>> data;
     private final Metadata metadata;
     private final InternalData internalData;
+    private final Map<String, Integer> idToIDX;
 
-    Dataframe() {
+    public Dataframe() {
         this.data = new ArrayList<>(new ArrayList<>());
         this.metadata = new Metadata();
         this.internalData = new InternalData();
+        this.idToIDX = new HashMap<>();
     }
 
     Dataframe(List<List<Value>> data, Metadata metadata) {
         this.data = new ArrayList<>(data);
         this.metadata = metadata;
         this.internalData = new InternalData();
+        this.idToIDX = new HashMap<>();
     }
 
     Dataframe(List<List<Value>> data, Metadata metadata, InternalData internalData) {
         this.data = new ArrayList<>(data);
         this.metadata = metadata;
         this.internalData = internalData;
+        this.idToIDX = new HashMap<>();
+        calculateIndexHashes();
```

Comment:
Minor comment (this is going to always trigger the `calculateIndexHashes()` method but it is going to be a no-op when `internalData` is empty)
```suggestion
    public Dataframe() {
        this(new ArrayList<>(new ArrayList<>()), new Metadata(), new InternalData());
    }

    Dataframe(List<List<Value>> data, Metadata metadata) {
        this(new ArrayList<>(data), metadata, new InternalData());
    }

    Dataframe(List<List<Value>> data, Metadata metadata, InternalData internalData) {
        this.data = new ArrayList<>(data);
        this.metadata = metadata;
        this.internalData = internalData;
        this.idToIDX = new HashMap<>();
        calculateIndexHashes();
```